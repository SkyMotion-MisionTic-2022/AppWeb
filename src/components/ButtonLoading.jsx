import React from 'react';
import ReactLoading from 'react-loading';

const ButtonLoading = ({ disabled, loading, text,  onClick = () => {} }) => {
    return (
        <button
            data-testid='button-loading'
            onClick={onClick}
            disabled={disabled}
            type='submit'
            className='bg-indigo-600 text-white font-bold text-lg py-2 px-1  rounded-xl hover:bg-indigo-500 shadow-md my-1 disabled:opacity-50 disabled:bg-gray-700'
        >
            {loading ? <ReactLoading data-testid="loading-in-button" type='spin' height={30} width={30} /> : text}
        </button>
    );
};

export default ButtonLoading;
