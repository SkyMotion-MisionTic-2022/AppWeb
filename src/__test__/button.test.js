import React from "react";
import ButtonLoading from 'components/ButtonLoading';
import { render, screen, cleanup } from '@testing-library/react';

afterEach(cleanup);

it('renders ok', () => {
    render(<ButtonLoading text='volver' loading={false} disasbled={false} />);
    expect(screen.getByTestId('button-loading')).toBeInTheDocument();
});

it('Mostrar texto cuando no este cargando', () => {
    render(<ButtonLoading text='volver' loading={false} disasbled={false} />);
    expect(screen.getByTestId('button-loading')).toHaveTextContent('volver');
});

it('No mostrar texto cuando este cargando', () => {
    render(<ButtonLoading text='volver' loading={true} disasbled={false} />);
    expect(screen.getByTestId('button-loading')).not.toHaveTextContent('volver');
});

it('Mostrar ReactLoading cuando estÃ© cargando', () => {
    render(<ButtonLoading text='volver' loading={true} disabled={false} />);
    expect(screen.getByTestId('loading-in-button')).toBeInTheDocument();
});

it('Deshabilitar cuando se pase el prop', () => {
    render(<ButtonLoading text='volver' loading={true} disabled={true} />);
    expect(screen.getByTestId('button-loading')).toHaveAttribute('disabled');
});

it('Cargar svg cuando el loading este activado', () => {
    render(<ButtonLoading text='volver' loading={true} disabled={true} />);
    expect(screen.getByTestId('button-loading')).toMatchSnapshot();
});
