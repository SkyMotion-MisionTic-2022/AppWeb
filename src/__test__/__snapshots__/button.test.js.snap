// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cargar svg cuando el loading este activado 1`] = `
<button
  class="bg-indigo-600 text-white font-bold text-lg py-2 px-1  rounded-xl hover:bg-indigo-500 shadow-md my-1 disabled:opacity-50 disabled:bg-gray-700"
  data-testid="button-loading"
  disabled=""
  type="submit"
>
  <div
    data-testid="loading-in-button"
    style="fill: #fff; height: 30px; width: 30px;"
  >
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
        opacity=".25"
      />
      
  
      <path
        d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
      >
        
    
        <animatetransform
          attributeName="transform"
          dur="0.8s"
          from="0 16 16"
          repeatCount="indefinite"
          to="360 16 16"
          type="rotate"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</button>
`;
